pytest_fixtures/component/provision_pxe.py

```diff
@@ -0,0 +1,227 @@
+    # TODO: investigate DNS setup issue on Satellite,
+    # we might need to set up Sat's DNS server as the primary one on the Sat host
+    provisioning_upstream_dns_primary = (
+        broker_data_out.provisioning_upstream_dns.pop()
+    )  # There should always be at least one upstream DNS
+    provisioning_upstream_dns_secondary = (
+        broker_data_out.provisioning_upstream_dns.pop()
+        if len(broker_data_out.provisioning_upstream_dns)
+        else None
+    )
+
+    domain = sat.api.Domain(
+        location=[module_location],
+        organization=[module_org_with_manifest],
+        dns=module_provisioning_capsule.id,
+        name=provisioning_domain_name,
+    ).create()
+
+    subnet = sat.api.Subnet(
+        location=[module_location],
+        organization=[module_org_with_manifest],
+        network=str(provisioning_network.network_address),
+        mask=str(provisioning_network.netmask),
+        gateway=broker_data_out.provisioning_gw_ipv4,
+        from_=broker_data_out.provisioning_host_range_start,
+        to=broker_data_out.provisioning_host_range_end,
+        dns_primary=provisioning_upstream_dns_primary,
+        dns_secondary=provisioning_upstream_dns_secondary,
+        boot_mode='DHCP',
+        ipam='DHCP',
+        dhcp=module_provisioning_capsule.id,
+        tftp=module_provisioning_capsule.id,
+        template=module_provisioning_capsule.id,
+        dns=module_provisioning_capsule.id,
+        httpboot=module_provisioning_capsule.id,
+        discovery=module_provisioning_capsule.id,
+        remote_execution_proxy=[module_provisioning_capsule.id],
+        domain=[domain.id],
+    ).create()
+
+    hostgroup = sat.api.HostGroup(
+        organization=[module_org_with_manifest],
+        location=[module_location],
+        architecture=default_architecture,
+        domain=domain,
+        content_source=module_provisioning_capsule.id,
+        content_view=module_default_org_view,
+        kickstart_repository=module_provisioning_rhel_content.ksrepo,
+        lifecycle_environment=module_lce_library,
+        root_pass=host_root_pass,
+        operatingsystem=module_provisioning_rhel_content.os,
+        ptable=default_partitiontable,
+        subnet=subnet,
+        pxe_loader=pxe_loader,
+        group_parameters_attributes=[  # assign AK in order the hosts to be subscribed
+            {
+                'name': 'remote_execution_ssh_keys',
+                'parameter_type': 'string',
+                'value': settings.provisioning.host_ssh_key_pub,
+            },
+            {
+                'name': 'kt_activation_keys',
+                'parameter_type': 'string',
+                'value': module_provisioning_rhel_content.ak.name,
+            },
+        ],
+    ).create()
+
+    return Box(sat=sat, hostgroup=hostgroup, subnet=subnet)
+
+
+@pytest.fixture(scope='module')
+def module_ssh_key_file():
+    (_, layout) = mkstemp(text=True)
```

Comment:
The tuple syntax is implied, so not required here for unpacking
```suggestion
    _, layout = mkstemp(text=True)
```

Reply:
changed